### PR TITLE
Update release instructions to follow newest GSG

### PR DESCRIPTION
**Pull request by hurryabit**
We've lately integrated the codegens into `daml start`, which made the GSG significantly shorter in turn. Reflect these changes in the release instructions. Also reword some parts where I found the instructions ambiguous on first reading.

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

---
This change is [Reviewable](https://reviewable.io/reviews/digital-asset/daml/7595)

### CODE REVIEW

File: release/RELEASE.md [hunk 1]
```diff
@@ -125,31 +125,32 @@ patches we backport to the 1.0 release branch).
...
-       1. `daml build`
-
-       1. `daml codegen js .daml/dist/create-daml-app-0.1.0.dar -o ui/daml.js`
-
        1. `daml start`
 
     1. In a new terminal, from the `ui` folder:
 
-       1. `npm install`
+       1. `npm install --frozen-lockfile`
```
Comment (cocreature) [inline release/RELEASE.md:132]: I don’t think this makes sense. This is the first call to `npm install`, you are generating the lockfile here.

File: release/RELEASE.md [hunk 2]
```diff
@@ -125,31 +125,32 @@ patches we backport to the 1.0 release branch).
...
 
        1. `npm start`
 
     1. Open two browser windows (you want to see them simultaneously ideally) at `localhost:3000`.
 
     1. Log in as `Alice` in the first window, log in as `Bob` in the second window.
 
-    1. Make `Alice` follow `Bob`. Verify that `Bob` appears in the
+    1. In the first window, where you are logged in as`Alice`,
```
Comment (cocreature) [inline release/RELEASE.md:140]: Seems a bit excessive but if it helps :shrug:

Comment (garyverhaegen-da) [inline release/RELEASE.md:140]:
```suggestion
    1. In the first window, where you are logged in as `Alice`,
```

**Review by garyverhaegen-da — APPROVED** [review]
Thanks!

**Review by cocreature — APPROVED** [review]
lgtm apart from the `npm install` change

File: release/RELEASE.md (hunk 1 above)
Comment (hurryabit) [reply release/RELEASE.md:132]: You're totally right. I was blindly following the GSG but apparently the GSG is wrong in this regard.

**Review by nemanja-da — APPROVED** [review]
tyvm 🙂

**Pull request merged**